share/completions/git.fish

```diff
@@ -1355,17 +1369,56 @@ complete -f -c git -n '__fish_git_using_command gc' -l keep-largest-pack -d 'Ign
 
 ### grep
 complete -c git -n __fish_git_needs_command -a grep -d 'Print lines matching a pattern'
-# TODO options
+complete -f -c git -n '__fish_git_using_command grep' -l cached -d 'Search blobs registered in the index file'
+complete -f -c git -n '__fish_git_using_command grep' -l no-index -d 'Search files in the current directory not managed by Git'
+complete -f -c git -n '__fish_git_using_command grep' -l untracked -d 'Search also in untracked files'
+complete -f -c git -n '__fish_git_using_command grep' -l no-exclude-standard -d 'Also search in ignored files by not honoring the .gitignore mechanism'
+complete -f -c git -n '__fish_git_using_command grep' -l exclude-standard -d 'Do not pay attention to ignored files specified via the .gitignore mechanism'
```

Comment:
Considering that fish will not show this one next to --no-exclude-standard (because completions are sorted), this should be slightly clearer:

```suggestion
complete -f -c git -n '__fish_git_using_command grep' -l exclude-standard -d 'Do not search ignored files specified via the .gitignore mechanism'
```